assets/extensions/extension-actions.js

```diff
@@ -44,19 +47,37 @@ export default ExtensionActions;
 /**
  * Get extension actions array.
  *
- * @param {Object} extension Extension object.
+ * @param {Object} extension           Extension object.
+ * @param {string} componentInProgress Which component has caused an update.
  *
  * @return {Array|null} Array of actions, or null if it's not a valid extension.
  */
-export const getExtensionActions = ( extension ) => {
+export const getExtensionActions = ( extension, componentInProgress ) => {
 	if ( ! extension.product_slug ) {
 		return null;
 	}
 
 	let buttonLabel = '';
+	let buttonAction = () => {};
 
-	if ( extension.has_update ) {
+	if ( componentInProgress === extension.product_slug ) {
```

Comment:
I think that this part of the code needs a bit of cleanup. But since we will need to add extra cases here when we implement 'Install' and/or 'Renew', I left it as it is for now.